src/Examples/Sorting/Sequential/InsertionSort.agda

```diff
@@ -113,37 +113,64 @@ sort/correct (x ∷ xs) u =
   ) , sorted-x∷xs'
 
 sort/cost : cmp (Π (list A) λ _ → meta ℂ)
-sort/cost l = length l ²
+sort/cost l = λ _ → length l ²
+
+Modal : (⋄ : tp pos → tp pos) (A : tp pos) → Set
+Modal ⋄ A = val (⋄ A) ↔ val A
+
+postulate
+  lemma : (A : tp pos) (e : cmp (F A)) (v : val A) → ◯ (e ≡ ret v) →
+    (X : tp neg) (f : val A → cmp X) →
+    bind X e f ≡ bind X e (const (f v))
+
+  lemma' : (A : tp pos) (h : Modal ◯⁺_ A) {e : ◯ (val A)} (u : ext) → Inverse.to h e ≡ e u
+
+  list-modal : Modal ◯⁺_ (list A)
```

Comment:
Is there a (cleaner) way to avoid this?

Reply:
Well, the problem is that it is simply not true unless `A` is modal.

Reply:
Heh, of course! Figured I'd test it out here, though, before propagating that assumption.